projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -237,77 +248,38 @@ def _store(self, cmd):
-                             ', "target": ' + str(qb_pos) + \
-                             ', "type": "c' + gatetxt + '"}, '
-            return
-        if gate == X and get_control_count(cmd) == 1:
-            ctrl_pos = cmd.control_qubits[0].id
-            qb_pos = cmd.qubits[0][0].id
-            self._circuit += '{"control": ' + str(ctrl_pos) + \
-                             ', "target": ' + str(qb_pos) + \
-                             ', "type": "cnot"}, '
-            return
-        if gate == Swap and get_control_count(cmd) == 1:
-            ctrl_pos = cmd.control_qubits[0].id
-            qb_pos0 = cmd.qubits[0][0].id
-            qb_pos1 = cmd.qubits[1][0].id
-            self._circuit += '{"targets": [' + \
-                str(qb_pos0) + ', ' + str(qp_pos1) + \
-                ' ], \
-                "control": ' + str(ctrl_pos) + \
-                '"type": "cswap"}, '
-            return
-        if gate == X and get_control_count(cmd) == 2:
-            ctrl_pos0 = cmd.control_qubits[0].id
-            ctrl_pos1 = cmd.control_qubits[1].id
-            qb_pos = cmd.qubits[0][0].id
-            self._circuit += '{"controls": [ ' + \
-                             str(ctrl_pos0) + ', ' + str(ctrl_pos1) + \
-                             ' ], \
-                                "target": ' + str(qb_pos) + \
-                             ', "type": "ccnot"}, '
-            return
-        if get_control_count(cmd) == 0:
-            if isinstance(gate, (R, Rx, Ry, Rz)):
-                qb_pos = cmd.qubits[0][0].id
-                angle = gate.angle
-                gatetxt = self._gationary[gate.__str__().split('(')[0]]
-                self._circuit += '{"angle": ' + str(angle) + \
-                                 ', "target": ' + str(qb_pos) + \
-                                 ', "type": "' + gatetxt + '"}, '
-                return
-            if gate in (X, Y, Z, H, S, T, Sdag, Tdag, SqrtX):
-                qb_pos = cmd.qubits[0][0].id
-                gatetxt = self._gationary[gate.__str__()]
-                self._circuit += '{"target": ' + str(qb_pos) + \
-                                 ', "type": "' + gatetxt + '"}, '
-                return
-            if gate == Swap:
-                qb_pos0 = cmd.qubits[0][0].id
-                qb_pos1 = cmd.qubits[1][0].id
-                self._circuit += '{"targets": [' + \
-                                 str(qb_pos0) + ', ' + str(qp_pos1) + \
-                                 '], \
-                                 "type": "swap"}, '
-                return
-            # TODO: Add unitary for the SV1 simulator as MatrixGate
-        if gate == Barrier:
-            return
+
+        json_cmd = {}
+
+        if num_controls == 2:
+            json_cmd['controls'] = [
+                cmd.control_qubits[0].id, cmd.control_qubits[1].id]
+        if num_controls == 1:
+            json_cmd['control'] = cmd.control_qubits[0].id
```

Comment:
Unfortunately this didn't work because the "else" includes num_controls = 1 and num_controls = 0, and in the latter case it  fails with cmd.control_qubits[0].id IndexError. In addition for num_controls = 0 there mus be no `['control']` in the json

I have replaced line 258 with your proposal `[qb.id for qb in cmd.control_qubits]`

Reply:
Then just do `elif num_controls == 1`